### PR TITLE
New command Set-DbaAgentJob

### PR DESCRIPTION
**Pull request by sanderstad**

### CODE REVIEW

File: functions/Set-DbaAgentJob.ps1 [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+                        $Job.EmailLevel = $EmailLevel
+                    }
+                    else {
+                        # Check if either the operator e-mail parameter is set or the operator is set in the job
+                        if (($EmailOperatorName.Length -ge 1) -or ($Job.OperatorToEmail.Length -ge 1)) {
+                            Write-Message -Message "Setting job e-mail level to $($EmailLevel)" -Level Verbose
+                            $Job.EmailLevel = $EmailLevel
+                        }
+                        else {
+                            Write-Message -Message "Cannot set e-mail level $EmailLevel without a valid e-mail operator name" -Warning
```
Comment (FriedrichWeinmann) [inline functions/Set-DbaAgentJob.ps1]: The parameter `-Warning` is deprecated, use `-Level Warning` instead

File: functions/Set-DbaAgentJob.ps1 [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+.PARAMETER SqlInstance
+SQL Server instance. You must have sysadmin access and server version must be SQL Server version 2000 or greater.
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. To use:
+$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter. 
+To connect as a different Windows user, run PowerShell as that user.
+
+.PARAMETER JobName
+The name of the job. Can be null if the the job id is being used.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: Hey Sander, all of our other commands use just -Database or -Job, etc. Please specify just -Job here

File: functions/Set-DbaAgentJob.ps1 [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+
+.PARAMETER NewName
+The new name for the job.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: This is fine for clarity

File: functions/Set-DbaAgentJob.ps1 [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+Disabled the job
+
+.PARAMETER Description
+The description of the job.
+
+.PARAMETER StartStepId
+The identification number of the first step to execute for the job.
+
+.PARAMETER CategoryName
+The category of the job.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: This should be Category

File: functions/Set-DbaAgentJob.ps1 [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+
+.PARAMETER OwnerLoginName
+The name of the login that owns the job.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: OwnerLogin is better as well probably.

File: functions/Set-DbaAgentJob.ps1 [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+Allowed values 0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always"
+The text value van either be lowercase, uppercase or something in between as long as the text is correct.
+
+.PARAMETER PageLevel
+Specifies when to send a page upon the completion of this job.
+Allowed values 0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always"
+The text value van either be lowercase, uppercase or something in between as long as the text is correct.
+
+.PARAMETER EmailOperatorName
+The e-mail name of the operator to whom the e-mail is sent when EmailLevel is reached.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: Yeah, same with just remove Name.

File: functions/Set-DbaAgentJob.ps1 [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+
+.PARAMETER NetsendOperatorName
+The name of the operator to whom the network message is sent.
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: And so on.

File: functions/Set-DbaAgentJob.ps1 [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+
+.PARAMETER PageOperatorName
```
Comment (potatoqualitee) [inline functions/Set-DbaAgentJob.ps1]: And so on.

**Comment by potatoqualitee** [comment]
Merci, and sorry I didn't catch this before

**Comment by potatoqualitee** [comment]
awesome work, @sanderstad! and tons of it. thank you! 🙇

**Pull request merged**